### PR TITLE
GitHub branching

### PR DESCRIPTION
**Pull request by ncphillips**
- [x] List Available Branches
- [x] Switch Branches
- [x] Create Branches
- [x] Add Confirmation dialogue before switching

image

### CODE REVIEW

File: packages/react-tinacms-github/src/toolbar-plugins/BranchSwitcherPlugin.tsx [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+  const [branches, setBranches] = React.useState<Branch[]>([])
+  React.useEffect(() => {
+    github
+      .getBranchList()
+      .then(branches => {
+        setBranches(branches)
+        setBranchStatus('loaded')
+      })
+      .catch(() => {
+        setBranchStatus('error')
```
Comment (jamespohalloran) [inline packages/react-tinacms-github/src/toolbar-plugins/BranchSwitcherPlugin.tsx:65]: It doesn't look like we're handling an error branchStatus currently

Comment (ncphillips) [reply packages/react-tinacms-github/src/toolbar-plugins/BranchSwitcherPlugin.tsx:65]: Indeed we are not.

@spbyrne want to add a little "We had trouble loading branches. Please refresh to try again."

**Comment by jamespohalloran** [comment]
Wondering if we need to scope the localStorage unsaved changes to workingRepo / branch
https://github.com/tinacms/tinacms/pull/1066#issuecomment-620620168

**Comment by ncphillips** [comment]
Possibly, but I think that's outside the scope of this PR. The localstorage caching is not done by these packages.

File: packages/react-tinacms-github/src/toolbar-plugins/BranchSwitcherPlugin.tsx (hunk 1 above)
Comment (spbyrne) [reply packages/react-tinacms-github/src/toolbar-plugins/BranchSwitcherPlugin.tsx:65]: Yeah good call.

Comment (spbyrne) [reply packages/react-tinacms-github/src/toolbar-plugins/BranchSwitcherPlugin.tsx:65]: Added 👍

**Pull request merged**